mongo/change_stream.go

```diff
@@ -457,20 +457,47 @@ func (cs *ChangeStream) ResumeToken() bson.Raw {
 }
 
 // Next gets the next result from this change stream. Returns true if there were no errors and the next
-// result is available for decoding.
+// result is available for decoding. Next blocks until an event is available for decoding or ctx expires.
+// If the given context expires during execution, the stream's error will be set and the change stream may be in an
+// invalid state and should be re-created. If Next returns false and an error occurred (i.e. cs.Err() != nil),
+// it must not be called again.
```

Comment:
What does `must not be called again` actually mean?   Consider changing to just say what the behavior is if you call it again.  Otherwise it sound more dire that it may be, like it panics or something.

Reply:
The issue is that we don't really know what the behavior may be. For the case of context cancellation, the change stream might be in some weird state where we have no idea what batch it's on because the stream might have been successfully resumed by the resume attempt. For the case of another error or the resume attempt failing, the subsequent call to `Next` will return false right away because the `err` field will be non-nil. I think it's safe to say "this is undefined, don't do it".